common/models/acl.js

```diff
@@ -377,30 +381,62 @@ module.exports = function(ACL) {
     }
   };
 
+  // NOTE Regarding ACL.isAllowed() and ACL.prototype.isAllowed()
+  // Extending existing logic, including from ACL.checkAccessForContext() method,
+  // ACL instance with missing property `permission` are not promoted to
+  // permission = ACL.DEFAULT config. Such ACL instances will hence always be
+  // inefective
+
+  /**
+   * Test if ACL's permission is ALLOW
+   * @param {String} permission The permission to test, expects one of 'ALLOW', 'DENY', 'DEFAULT'
+   * @param {String} defaultPermission The default permission to apply if not providing a finite one in the permission parameter
+   * @returns {Boolean} true if ACL permission is ALLOW
+   */
+  ACL.isAllowed = function(permission, defaultPermission) {
+    if (permission === ACL.DEFAULT) {
+      permission = defaultPermission || ACL.ALLOW;
+    }
+    return permission !== loopback.ACL.DENY;
+  };
+
+  /**
+   * Test if ACL's permission is ALLOW
+   * @param {String} defaultPermission The default permission to apply if missing in ACL instance
+   * @returns {Boolean} true if ACL permission is ALLOW
+   */
+  ACL.prototype.isAllowed = function(defaultPermission) {
+    return this.constructor.isAllowed(this.permission, defaultPermission);
+  };
+
   /**
    * Check if the request has the permission to access.
-   * @options {Object} context See below.
+   * @options {AccessContext} context An AccessContext instance.
    * @property {Object[]} principals An array of principals.
-   * @property {String|Model} model The model name or model class.
-   * @property {*} id The model instance ID.
+   * @property {Function} model The model class.
+   * @property {String} modelName The model name.
+   * @property {*} modelId The model instance ID.
    * @property {String} property The property/method/relation name.
-   * @property {String} accessType The access type:
-   *   READ, REPLICATE, WRITE, or EXECUTE.
-   * @param {Function} callback Callback function
+   * @property {String} accessType The access type: READ, REPLICATE, WRITE, or EXECUTE.
+   * @callback {Function} callback Callback function
+   * @param {String|Error} err The error object.
+   * @param {AccessRequest} result The resolved access request.
    */
```

Comment:
jsdoc edit, following definition in [AccessContext](https://github.com/strongloop/loopback/blob/master/lib/access-context.js#L11)